### PR TITLE
Add webUI acceptance test for editing user email

### LINKED ISSUE #137: WebUi acceptance tests for editing user email
#128 added acceptance tests for editing user attributes such as display name, password, groups.
But tests for editing email is still not implemented. That needs to be added.

### PR DESCRIPTION
**Pull request by dpakach**
Fixes #137 

Add webUI acceptance test for editing user email by the administrator.

### CODE REVIEW

File: tests/acceptance/features/lib/UsersPage.php [hunk 1]
```diff
@@ -658,6 +661,42 @@ public function changeUserPassword(Session $session, $user, $password) {
...
+				__METHOD__ .
+				" xpath $this->editEmailInputXpath " .
+				"could not find email field "
+			);
+		}
+		try {
+			$editEmailInput->focus();
+			$editEmailInput->setValue($password . "\n");
+			$this->waitForAjaxCallsToStartAndFinish($session);
+		} catch (StaleElementReference $e) {
```
Comment (skshetry) [inline tests/acceptance/features/lib/UsersPage.php]: Please explain why this catch is required here. Might help someone in the future.

File: tests/acceptance/features/webUIManageUsersGroups/editUsers.feature [hunk 2]
```diff
@@ -83,3 +83,11 @@ Feature: edit users
     And user "user0" should not belong to group "grp1"
     And user "user0" should not belong to group "grp2"
     And user "user0" should belong to group "grp3"
+
+  Scenario: Admin changes the email of the user
+    Given user "user0" has been created with default attributes
+    And the administrator has browsed to the users page
+    When the administrator changes the email of user "user0" to "new_email@oc.com" using the webUI
+    Then user "user0" should exist
```
[the remark this answers is not in the document]
Comment (dpakach) [reply tests/acceptance/features/webUIManageUsersGroups/editUsers.feature]: Yes. Just to be sure

File: tests/acceptance/features/lib/UsersPage.php [hunk 3]
```diff
@@ -658,6 +661,44 @@ public function changeUserPassword(Session $session, $user, $password) {
 		}
 	}
 
+	/**
+	 * @param Session $session
+	 * @param string $user
+	 * @param string $password
```
Comment (individual-it) [inline tests/acceptance/features/lib/UsersPage.php]:
```suggestion
	 * @param string $email
```

File: tests/acceptance/features/lib/UsersPage.php [hunk 4]
```diff
@@ -658,6 +661,44 @@ public function changeUserPassword(Session $session, $user, $password) {
...
+	 *
+	 * @return void
+	 */
+	public function changeUserEmail(Session $session, $user, $password) {
+		$userTr = $this->findUserInTable($user);
+		$editEmailBtn = $userTr->find("xpath", $this->editEmailBtnXpath);
+		if ($editEmailBtn === null) {
```
Comment (individual-it) [inline tests/acceptance/features/lib/UsersPage.php]: use `$this->assertElementNotNull()`

Comment (phil-davis) [reply tests/acceptance/features/lib/UsersPage.php]: There are quite a few places like this in the methods above. @individual-it should we refactor that stuff while cut-pasting here for this new method?

**Comment by dpakach** [comment]
Backport in https://github.com/owncloud/core/pull/34305

**Pull request merged**